### PR TITLE
[new-model] Add V2A: native MMAudio inference pipeline

### PR DESCRIPTION
**Pull request by kaiw7**
## Purpose

Add a native FastVideo video-to-audio (V2A) and text-to-audio (T2A) inference pipeline for MMAudio, starting with the `large_44k_v2` checkpoint.

The implementation follows FastVideo's component, pipeline-stage, preset, registry, and checkpoint-loading abstractions. It does not import the upstream MMAudio Python package at inference time and does not change the routing or behavior of existing T2V, I2V, or T2I pipelines.

## Changes

- Add `V2A` and `T2A` workload registration and MMAudio pipeline configuration/preset.
- Add native MMAudio transformer, DFN5B CLIP text/vision encoders, Synchformer visual encoder, 44.1 kHz audio VAE, and BigVGAN-v2 vocoder components.
- Add FastVideo pipeline stages for video preprocessing, multimodal conditioning, flow-matching denoising, audio decoding, and audio-only output.
- Support variable-duration V2A inference while retaining 8 seconds as the default duration.
- Reuse FastVideo component loading, offloading, scheduling, registry, and `VideoGenerator` execution paths.
- Add an official-to-FastVideo checkpoint conversion script and Diffusers-style component layout.
- Add a runnable V2A inference example and MMAudio-specific setup/conversion documentation.
- Add component-level and full-pipeline parity tests against the official MMAudio implementation.
- Keep official and converted model weights outside the Git repository.

## Test Plan

```bash
export VIDEO_PATH=/path/to/input.mp4

# Local MMAudio test suite
pytest -q tests/local_tests/mmaudio

# Opt-in real-weight official-vs-FastVideo pipeline parity
MMAUDIO_RUN_PIPELINE_PARITY=1 \
MMAUDIO_PARITY_VIDEO="$VIDEO_PATH" \
pytest -q \
  tests/local_tests/mmaudio/test_mmaudio_pipeline_parity.py::test_mmaudio_real_v2a_pipeline_waveform_parity \
  -s

# Real 10-second V2A inference
MMAUDIO_MODEL_PATH=converted_weights/mmaudio/large_44k_v2 \
python examples/inference/basic/basic_mmaudio.py \
  --video-path "$VIDEO_PATH" \
  --duration-seconds 10 \
  --output-path outputs_audio/mmaudio_10s.wav
```

## Test Results

<details>
<summary>Test output</summary>

```text
Local MMAudio suite: 18 passed, 1 skipped

Official-vs-FastVideo real-weight parity:
- video preprocessing: exact
- condition features and initial latent: exact
- first flow prediction: exact
- 25-step final latent: exact
- final 2-second waveform (89,088 samples): exact (atol=0, rtol=0)

Variable-duration inference:
- requested duration: 10 seconds
- output: 441,344 samples at 44.1 kHz (10.0078 seconds)
- result: passed

FastVideo default component/layerwise offload path:
- real one-step smoke test: passed
```

</details>

Validated on Python 3.12, PyTorch 2.12.0+cu126. 

## Checklist

- [ ] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests for my changes
- [x] I updated documentation if needed
- [x] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**

- [x] SSIM is not applicable to audio-only output; exact waveform parity was verified instead
- [ ] I updated the support matrix if adding a new model

## Remaining Before Merge

- Fix the currently failing `pre-commit` check and update the checklist above.
- Address or resolve the two open automated code-review comments.
- Add MMAudio to `docs/inference/support_matrix.md`, then update the checklist above.
- Obtain at least one approving review and pass FastCheck plus the Full Test Suite through the repository merge workflow.

### CODE REVIEW

**Comment by SolitaryThinker** [comment]
Maintainer validation updated at `72634a65b`.

- VAE state-dict keys and shapes are preserved.
- GB200 random-weight decoder parity passed in FP32 and BF16 (`atol=1e-6`, `rtol=1e-6`).
- Public `v1-44.pth` MD5 verified as `fab020275fa44c6589820ce025191600`; exact-head real-weight encode/decode parity passed (`1 passed` in 28.34 s, `atol=1e-5`, `rtol=1e-5`).
- Two 5-second, 25-step, seed-42 GB200 V2A generations completed end to end; both outputs are mono PCM16 at 44.1 kHz with 221,184 samples.
- Targeted pre-commit and combined-diff whitespace checks pass.

GitHub still reports a conflict with `main`; merge CI and rebase remain separate follow-ups.

**Comment by SolitaryThinker** [comment]
Rebased onto origin/main at a2bfc7cdb to resolve conflicts and pick up the current MiniMax H3/Kandinsky registry entries; no content changes (range-diff clean).

File: fastvideo/models/dits/mmaudio.py [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+                10000 ** (torch.arange(0, frequency_embedding_size, 2, dtype=torch.float32) / frequency_embedding_size)
+            )
+            self.register_buffer("freqs", (10000 / max_period) * freqs, persistent=False)
+
+    def timestep_embedding(self, timestep: torch.Tensor) -> torch.Tensor:
+        args = timestep[:, None].float() * self.freqs[None]
+        return torch.cat([torch.cos(args), torch.sin(args)], dim=-1)
+
+    def forward(self, timestep: torch.Tensor) -> torch.Tensor:
+        embedding = self.timestep_embedding(timestep).to(timestep.dtype)
```
[the remark this answers is not in the document]
Comment (SolitaryThinker) [reply fastvideo/models/dits/mmaudio.py]: Fixed in `20af6314a`: `TimestepEmbedder` now casts the sinusoidal embedding to the MLP weight dtype, and `fastvideo/tests/transformers/test_mmaudio.py` covers integer timesteps against the equivalent float input.

File: fastvideo/models/audio/mmaudio_vae.py [hunk 2]
```diff
@@ -0,0 +1,813 @@
...
+    0.9101,
+    0.7164,
+]
+
+
+def normalize(x: torch.Tensor, dim: int | list[int] | None = None, eps: float = 1e-4) -> torch.Tensor:
+    if dim is None:
+        dim = list(range(1, x.ndim))
+    norm = torch.linalg.vector_norm(x, dim=dim, keepdim=True, dtype=torch.float32)
+    norm = torch.add(eps, norm, alpha=np.sqrt(norm.numel() / x.numel()))
```
[the remark this answers is not in the document]
Comment (SolitaryThinker) [reply fastvideo/models/audio/mmaudio_vae.py]: The current head no longer uses `torch.add` in this path; the magnitude-preserving normalization uses tensor arithmetic directly. Resolving this outdated thread.

**Comment by SolitaryThinker** [comment]
Applied the requested follow-up in `20af6314a`: MMAudio timestep embeddings now cast to the MLP parameter dtype, with an integer-vs-float regression test, and the inference support matrix now lists the registered V2A/T2A entry plus the current local-conversion requirement. Local targeted pytest: 1 passed. Targeted pre-commit: passed. Fresh Buildkite #4780 is pending.

**Comment by SolitaryThinker** [comment]
Rebased onto `origin/main@e0d702dec` to pick up #1702’s Python 3.12/YAPF update; no semantic content changes (range-diff: four commits `=`, with the two `!` entries limited to documented mechanical YAPF overlaps in the loader/registry and Synchformer extraction).

**Review by SolitaryThinker — APPROVED** [review]
